lambda-packages/dns-validated-domain-identity-handler/src/record.ts

```diff
@@ -1,20 +1,20 @@
-import { Change } from "aws-sdk/clients/route53";
+import {Change, ResourceRecord} from "aws-sdk/clients/route53";
 
 const DEFAULT_VERIFICATION_RECORD_TTL = 1800; // 30 minutes
 
 export class Record {
-  public static forIdentity(domainName: string, token: string) {
-    return new this(`_amazonses.${domainName}`, "TXT", `"${token}"`);
+  public static forIdentity(domainName: string, records: ResourceRecord[]) {
+    return new this(`_amazonses.${domainName}`, "TXT", records);
   }
 
   public static forDKIM(domainName: string, token: string) {
-    return new this(`${token}._domainkey.${domainName}`, "CNAME", `${token}.dkim.amazonses.com`);
+    return new this(`${token}._domainkey.${domainName}`, "CNAME", [{Value: `${token}.dkim.amazonses.com`}]);
   }
 
   public constructor(
     public readonly name: string,
     public readonly type: "CNAME" | "TXT",
-    public readonly value: string,
+    public readonly records: ResourceRecord[],
```

Comment:
Is there any reason to use `ResourceRecord[]` instead of `string[]`?

Reply:
When fetching the existing Route53 records, they come back from the API as `ResourceRecord[]`. Instead of unwrapping them to `string[]` just for the constructor to convert them back to  `ResourceRecord[]`, I changed the constructor to work with `ResourceRecord[]` instead. Happy to change it back though if you would like.